itou/www/apply/forms.py

```diff
@@ -547,8 +547,10 @@ def _get_choices_for_administrativecriteria(self):
         return [(c.pk, c.name) for c in AdministrativeCriteria.objects.all()]
 
     def _get_choices_for_departments(self):
-        job_seekers = self.job_applications_qs.get_unique_fk_objects("job_seeker")
-        departments = {(user.department, DEPARTMENTS.get(user.department)) for user in job_seekers if user.department}
+        departments = set()
+        for job_seeker in self.job_applications_qs.get_unique_fk_objects("job_seeker"):
+            if job_seeker.department and DEPARTMENTS.get(job_seeker.department):
```

Comment:
```suggestion
            if job_seeker.department in DEPARTMENTS:
```
Plus simple et devrait marcher non ?

Reply:
En fait t'aurais même pu garder un one liner : `departments = {(user.department, DEPARTMENTS.get(user.department)) for user in job_seekers if user.department in DEPARTMENTS}` un peu long toutefois, donc tu as le choix.

Reply:
et un test ? ^^ /jeanmichelrelou

Reply:
> En fait t'aurais même pu garder un one liner : `departments = {(user.department, DEPARTMENTS.get(user.department)) for user in job_seekers if user.department in DEPARTMENTS}` un peu long toutefois, donc tu as le choix.

en effet, c'est plus sexy ! merci !

Reply:
> et un test ? ^^ /jeanmichelrelou

On pourrait en effet tester tous les cas tordus.